hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/enumerator/StaticHoodieEnumerator.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.source.enumerator;
+
+import org.apache.hudi.source.split.HoodieSourceSplit;
+import org.apache.hudi.source.split.SplitProvider;
+
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+
+/**
+ *  Static Hoodie split enumerator that only handles with a bounded number of hudi commits.
+ */
+public class StaticHoodieEnumerator extends AbstractHoodieEnumerator {
+  private final SplitProvider provider;
+
+  public StaticHoodieEnumerator(
+      SplitEnumeratorContext<HoodieSourceSplit> enumeratorContext, SplitProvider provider) {
+    super(enumeratorContext, provider);
+    this.provider = provider;
+  }
+
+  @Override
+  public void start() {
+    super.start();
+  }
+
+  @Override
+  protected boolean shouldWaitForMoreSplits() {
+    return false;
+  }
+
+  @Override
+  public HoodieEnumeratorState snapshotState(long checkpointId) {
```

Comment:
Seems not needed, because it's same with `snapshotState` in `AbstractHoodieEnumerator`